File: TransportesBackend/DTOs/AlmacenDTO.cs
using System;

namespace TransportesBackend.DTOs
{
    public class AlmacenDTO
    {
        // Ocultamos explícitamente el 'DeletedAt' y el 'DireccionId' original
        public string Id { get; set; }
        public string Nombre { get; set; }
        
        // Aplanamos la relación con la tabla 'Direccion'
        public string DireccionCompleta { get; set; }
        public string DireccionId { get; set; } 
        /* Ya no hacen falta pues estos pertenecen a direccion no a almacen
        public string Ciudad { get; set; } 
        public string Cp { get; set; } 
        public string Provincia { get; set; } 
        public string Pais { get; set; } 
        */
    }
}

File: TransportesFrontend/src/app/almacenes/almacen.dto.ts
// Fíjate: No hay 'DeletedAt'. Es el contrato limpio y seguro.
export interface AlmacenDTO {
    id: string;
    nombre: string;
    direccionId: string;
    direccionCompleta: string;
    /* Ya no hacen falta pues estos pertenecen a direccion no a almacen
    ciudad: string;
    cp: string;
    provincia: string;
    pais: string;
    */
}

// El DTO de Escritura (Reflejo exacto de tu C#)
export interface CreateAlmacenDTO {
  nombre: string;
  direccionId: string;
  /* Ya no hacen falta pues estos pertenecen a direccion no a almacen
  ciudad: string;
  cp: string;
  provincia: string;
  pais: string;
  */
}

export interface DireccionDTO {
  id: string;
  textoMostrar: string; // lo que se muestra en el select
  calle: string;
  ciudad: string;
  provincia: string;
  cp: string;
  pais: string;
}

export interface UpdateAlmacenDTO {
  nombre: string;
  direccionId: string;
  /* Ya no hacen falta pues estos pertenecen a direccion no a almacen
  ciudad: string;
  cp: string;
  provincia: string;
  pais: string;
  */
}